src/api.cc

```diff
@@ -0,0 +1,149 @@
+#include "aktualizr-lite/api.h"
+
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include "helpers.h"
+#include "libaktualizr/config.h"
+#include "liteclient.h"
+
+AkliteClient::AkliteClient(const std::vector<boost::filesystem::path>& config_dirs) {
+  Config config(config_dirs);
+  config.telemetry.report_network = !config.tls.server.empty();
+  config.telemetry.report_config = !config.tls.server.empty();
+  client_ = std::make_unique<LiteClient>(config, nullptr);
+  client_->finalizeInstall();
+}
+
+static bool compareTargets(const TufTarget& a, const TufTarget& b) { return a.Version() < b.Version(); }
+
+CheckInResult AkliteClient::CheckIn() const {
+  if (!configUploaded_) {
+    client_->reportAktualizrConfiguration();
+    configUploaded_ = true;
+  }
+  client_->reportNetworkInfo();
+  client_->reportHwInfo();
+
+  CheckInResult result(CheckInResult::Status::Ok);
+
+  Uptane::HardwareIdentifier hwidToFind(client_->config.provision.primary_ecu_hardware_id);
+
+  LOG_INFO << "Refreshing Targets metadata";
+  const auto rc = client_->updateImageMeta();
+  if (!std::get<0>(rc)) {
+    LOG_WARNING << "Unable to update latest metadata, using local copy: " << std::get<1>(rc);
+    if (!client_->checkImageMetaOffline()) {
+      LOG_ERROR << "Unable to use local copy of TUF data";
+      result.status = CheckInResult::Status::Failed;
+      return result;
+    }
+    result.status = CheckInResult::Status::OkCached;
+  }
+
+  for (const auto& t : client_->allTargets()) {
+    int ver = 0;
+    try {
+      ver = std::stoi(t.custom_version(), nullptr, 0);
+    } catch (const std::invalid_argument& exc) {
+      LOG_ERROR << "Invalid version number format: " << t.custom_version();
+      ver = -1;
+    }
+    if (!target_has_tags(t, client_->tags)) {
+      continue;
+    }
+    for (const auto& it : t.hardwareIds()) {
+      if (it == hwidToFind) {
+        result.targets_.emplace_back(t.filename(), t.sha256Hash(), ver, t.custom_data());
+        break;
+      }
+    }
+  }
+
+  std::sort(result.targets_.begin(), result.targets_.end(), compareTargets);
+  return result;
+}
+
+boost::property_tree::ptree AkliteClient::GetConfig() const {
+  std::stringstream ss;
+  ss << client_->config;
+
+  boost::property_tree::ptree pt;
+  boost::property_tree::ini_parser::read_ini(ss, pt);
+  return pt;
+}
+
+TufTarget AkliteClient::GetCurrent() const {
+  auto current = client_->getCurrent();
+  int ver = -1;
+  try {
+    ver = std::stoi(current.custom_version(), nullptr, 0);
+  } catch (const std::invalid_argument& exc) {
+    LOG_ERROR << "Invalid version number format: " << current.custom_version();
+  }
+  return TufTarget(current.filename(), current.sha256Hash(), ver, current.custom_data());
+}
+
+static InstallResult install(LiteClient& client, const Uptane::Target& t) {
+  client.logTarget("Installing: ", t);
+
+  auto rc = client.install(t);
```

Comment:
Just wondering why we return `data::ResultCode::Numeric`  instead of `data::InstallationResult` in `LiteClient::install`, including explanation of the installation failure would be really useful. :)

Reply:
i agree, i'm just trying to not turn this PR into an effort for fixing the liteclient API. having this API in place should make changing the liteclient signature easier